lib/Runtime/Library/ScriptFunction.cpp

```diff
@@ -753,7 +753,7 @@ namespace Js
                 }
             }
 
-            if (!isShutdown && unregisteredInlineCacheCount > 0 && !scriptContext->IsClosed())
+            if (unregisteredInlineCacheCount > 0)
```

Comment:
If you see above, we call `RemoveFromInvalidationList()` only if `!isShutdown` and `!scriptContext->IsClosed` which means `unregisteredInlineCacheCount  > 0` only if this 2 conditions are met. So the check here was redundant.